src/LinuxTracing/LibunwindstackMultipleOfflineAndProcessMemoryTest.cpp

```diff
@@ -0,0 +1,170 @@
+
+  ASSERT_EQ(read_count, 3);
+  EXPECT_EQ(destination[0], 0x20);
+  EXPECT_EQ(destination[1], 0x30);
+  EXPECT_EQ(destination[2], 0x40);
+}
+
+TEST(LibunwindstackMultipleOfflineAndProcessMemory, ReadFromSecondStackSlice) {
+  constexpr uint64_t kStartAddress1 = 0xADD8E55;
+  std::vector<char> bytes1{0x01, 0x10, 0x20, 0x30, 0x40};
+  StackSliceView stack_slice1(kStartAddress1, bytes1.size(), bytes1.data());
+
+  constexpr uint64_t kStartAddress2 = 0xABCDEF;
+  std::vector<char> bytes2{0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17};
+  StackSliceView stack_slice2(kStartAddress2, bytes2.size(), bytes2.data());
+
+  std::shared_ptr<unwindstack::Memory> sut =
+      LibunwindstackMultipleOfflineAndProcessMemory::CreateWithoutProcessMemory(
+          {stack_slice1, stack_slice2});
+
+  std::array<char, 3> destination{};
+  size_t read_count = sut->Read(kStartAddress2 + 2, destination.data(), 3);
+
+  ASSERT_EQ(read_count, 3);
+  EXPECT_EQ(destination[0], 0x13);
+  EXPECT_EQ(destination[1], 0x14);
+  EXPECT_EQ(destination[2], 0x15);
+}
+
+TEST(LibunwindstackMultipleOfflineAndProcessMemory,
+     RequestingToReadWithPartialIntersectionReturnsZero) {
+  constexpr uint64_t kStartAddress = 0xADD8E55;
+  std::vector<char> bytes{0x01, 0x10, 0x20, 0x30, 0x40};
+  StackSliceView stack_slice(kStartAddress, bytes.size(), bytes.data());
+
+  std::shared_ptr<unwindstack::Memory> sut =
+      LibunwindstackMultipleOfflineAndProcessMemory::CreateWithoutProcessMemory({stack_slice});
+
+  std::array<char, 3> destination{};
+  destination.fill(0x11);
+  size_t read_count = sut->Read(kStartAddress - 1, destination.data(), 3);
+
+  ASSERT_EQ(read_count, 0);
+  EXPECT_EQ(destination[0], 0x11);
+  EXPECT_EQ(destination[1], 0x11);
+  EXPECT_EQ(destination[2], 0x11);
+}
+
+TEST(LibunwindstackMultipleOfflineAndProcessMemory,
+     RequestingToReadUnknownMemoryWithoutProcessReturnsZero) {
+  constexpr uint64_t kStartAddress = 0xADD8E55;
+  std::vector<char> bytes{0x01, 0x10, 0x20, 0x30, 0x40};
+  StackSliceView stack_slice(kStartAddress, bytes.size(), bytes.data());
+
+  std::shared_ptr<unwindstack::Memory> sut =
+      LibunwindstackMultipleOfflineAndProcessMemory::CreateWithoutProcessMemory({stack_slice});
+
+  std::array<char, 3> destination{};
+  destination.fill(0x11);
+  size_t read_count = sut->Read(0xFE, destination.data(), 3);
+
+  ASSERT_EQ(read_count, 0);
+  EXPECT_EQ(destination[0], 0x11);
+  EXPECT_EQ(destination[1], 0x11);
+  EXPECT_EQ(destination[2], 0x11);
+}
+
+TEST(LibunwindstackMultipleOfflineAndProcessMemory,
+     ReadFromCompleteMemoryEvenIfPartiallyIntersectsWithOtherStackSlice) {
+  constexpr uint64_t kStartAddress1 = 0xADD8E55;
+  std::vector<char> bytes1{0x01, 0x10, 0x20, 0x30, 0x40};
+  StackSliceView stack_slice1(kStartAddress1, bytes1.size(), bytes1.data());
+
+  constexpr uint64_t kStartAddress2 = kStartAddress1 - 2;
+  std::vector<char> bytes2{0x0a, 0x0b, 0x0c, 0x0d, 0x0f, 0x10, 0x01};
```

Comment:
It is intended that the bytes don't match the ones in `bytes1`, right? I mean, it doesn't matter for the test so I don't mind, I'm mentioning it because they were matching before.

Reply:
Yeah, that was intentionally for the test. Before, I wanted to make it look more realistic, but I think if we use different bytes here, it reduces the risk of accidentally reading from the wrong memory.